clients/js/src/compat.ts

```diff
@@ -653,42 +640,7 @@ export async function fetchRuntimePublicKey(
         ) => Promise<any> | void;
       };
 
-      // For Truffle, turn a callback into an synchronous call
```

Comment:
Did you test the removal of this with Brave and Metamask? EIP-1193 is "Ethereum Provider JavaScript API", so this is not just truffle.

Reply:
Test flow of using Metamask on Brave with our example through the Hardhat [boilerplate](https://docs.oasis.io/dapp/sapphire/browser/) right?

That flow does work, but now I'm suspicious of my local dev and testing. Good callout. Lemme keep digging. 🤔

Reply:
K. I see the error now. We had another fallback mechanism, but I think the client should be EIP-1193 compliant as you corrected! 👍